stdlib/public/core/FloatingPoint.swift

```diff
@@ -330,8 +330,8 @@ public protocol FloatingPoint: SignedNumeric, Strideable, Hashable
 
   /// The [mathematical constant π][wiki], approximately equal to 3.14159.
   /// 
-  /// The π constant is equivalent to 180°, or half a revolution of a circle,
-  /// when used as an angle measurement in radians.
+  /// π is equivalent to 180°, or a half turn, when measuring an angle in
```

Comment:
Better and better! Grammatically, though, there remain a few issues:

- It's hard to say that pi is _equivalent_ to a half-turn, as one is a dimensionless value in radians and the other is, well, a turn—we wouldn't tell someone to "make a pi" instead of "make a half-turn" when we ask them to turn around. Perhaps it could be said that pi _denotes_ a half-turn. This problem would be obviated altogether with my suggestion to omit this entirely. No need to mention both 180° and a half-turn: the target audience almost certainly knows what angle is meant by 180°, and if they don't, this documentation isn't the place to teach them.

- Writing "in radians" at the end means that the sentence, read strictly, claims that "pi radians" is equivalent to "180° radians." However, we do not ever say "180° radians": degrees and radians are used mutually exclusively. Hence why I suggested writing "radians" after the Greek letter.

- It's certainly not wrong, but starting a paragraph with a lowercase (Greek) letter is generally best avoided _where it's possible to rephase_, according to most style guides. This is why I suggested moving "when measuring an angle" to the front of the sentence.

Reply:
> It's hard to say that pi is equivalent to a half-turn, as one is a dimensionless value in radians and the other is, well, a turn

[Angle is also used to designate the measure of an angle or of a rotation.](https://en.wikipedia.org/wiki/Angle#:~:text=Angle%20is%20also%20used%20to%20designate%20the%20measure%20of%20an%20angle%20or%20of%20a%20rotation.) So I think it's quite reasonable to compare a 180° angle with a half-turn, since both describe the same rotation. I wished to include this segment because equivalence to 180 degrees is a minor incidental fact, useful for conversions, but not core to the matter at hand. The actual message I wish to communicate is what angle π describes, and 180° does not really answer that - it just points the reader to the definition of degrees and expects them to do some mental math to figure out what it means in practice. A half-turn is a self-contained statement. This is a minor point, but since we're being pedantic I think it's worth taking into account.

> Writing "in radians" at the end means that the sentence, read strictly, claims that "pi radians" is equivalent to "180° radians."

I honestly don't see how "When measuring an angle in radians, π is equivalent to 180°." avoids this problem.

> It's certainly not wrong, but starting a paragraph with a lowercase (Greek) letter is generally best avoided where it's possible to rephase, according to most style guides.

This is partly why I originally wrote _the constant π_ instead. I feel like I'm, er, going in circles, here.

Anyway, not unwilling to change the current writing but I'm unconvinced your suggestion is strictly better.

Reply:
> So I think it's quite reasonable to compare a 180° angle with a half-turn, since both describe the same rotation.

I don't understand the sentence that "angle is used to designate the measure of an angle." I absolutely agree that it's reasonable to make a _comparison_ between an angle and a rotation. The problem I'm pointing out with the sentence is that it is stating that a dimensionless number, π, is equivalent to a half-turn, which is an angle—and π is no more an angle than is _e_ or 42.

> The actual message I wish to communicate is what angle π describes

I think that's great, and I like the verb "describe" here! Let's keep the sentence focused on that then: "When measuring angles in radians, π describes a half-turn." Pick either the half-turn or the 180°; otherwise, you're basically explaining to the reader that 180° is a half-turn, which is neither relevant to the definition of π nor news to the intended audience.

> I honestly don't see how "When measuring an angle in radians, π is equivalent to 180°." avoids this problem.

Placed as shown, the prepositional phrase ("when measuring an angle in radians") is adjectival, describing the subject of the sentence ("π"). Placed at the end, if read as an adjectival phrase, it would modify "180°"; even if one reads it as an adverbial phrase, it has the same issue of describing "180°" as something one could obtain "when measuring an angle in radians."

> This is partly why I originally wrote the constant π instead. I feel like I'm, er, going in circles, here.

Sure, I am fully in support of that!

Reply:
>  The problem I'm pointing out with the sentence is that it is stating that a dimensionless number, π, is equivalent to a half-turn, which is an angle—and π is no more an angle than is e or 42.

But both _e_ and 42 **can** be interpreted as angle measurements in radians. Being dimensionless, a radian is effectively just a real number and the radian unit exists only for adding context. I think your dispute is that the measurement of an angle is interchangeable with the angle itself, in a sentence. I'd argue that usage has made that acceptable, as people will say "do a 360" to mean doing a full turn. In fact, you yourself used an angle this way in the next paragraph at "explaining to the reader that 180° is a half-turn".

> otherwise, you're basically explaining to the reader that 180° is a half-turn, which is neither relevant to the definition of π nor news to the intended audience.

I agree that "180° is a half-turn" is not relevant to the definition of π, but I thought we agreed that defining π is not the point? Both facts that π is equivalent to 180° and that π is equivalent to a half-turn are relevant, for different reasons. It just so happens that they are related by transitivity. I'm happy to remove the 180° part, though, if you find it redundant.

> Placed as shown, the prepositional phrase ("when measuring an angle in radians") is adjectival, describing the subject of the sentence ("π"). Placed at the end, if read as an adjectival phrase, it would modify "180°"; even if one reads it as an adverbial phrase, it has the same issue of describing "180°" as something one could obtain "when measuring an angle in radians."

Fair, I guess English is a more complicated language than I ever gave it credit for. I'll concede the point.

Reply:
Thanks for your patience. I hope the current version is suitable.

Reply:
Thank _you_ for sticking with it! I think this version gets at your point with maximum clarity and concision.